canonical-web-and-design/ubuntu.com
# Write cypress test for /contact-us page forms

## Done

- Write cypress test for forms on /contact-us page
- move forms tests to `Cypress tests / main` workflow that runs only on push to main (effectively on after merge).

## QA

1. Create `cypress.env.json` file 
2. Copy and paste below (Reach out to me if you don't have keys) 
`{
    "UBUNTU_USERNAME": "",
    "UBUNTU_PASSWORD": "",
    "MARKETO_CLIENT_ID": "",
    "MARKETO_CLIENT_SECRET": "",
    "MARKETO_AUTHORISED_USER": "",
    "MARKETO_TOKEN": ""
}`
3. Run the site using the command `./run serve` or `dotrun`
4. Use`yarn run cypress open --config-file tests/cypress/cypress.json --config baseUrl=http://0.0.0.0:8001/` to run cypress test
5. Click `forms_spec.js`
6. Check if tests pass. 

## Issue / Card

Fixes #[4440](https://github.com/canonical-web-and-design/web-squad/issues/4440)
